Enhance pbkdf2 password hashing (#24132)
prestodb/presto: presto-docs/src/main/sphinx/security/password-validation.rst

===========================

> Password Validation Changes

Reviewer: ```suggestion
Password Validation
```
Also, please change the length of the formatting header lines above and below to match the length of this line after editing.

===========================

Overview
--------
`PBKDF2WithHmacSHA1` is no longer considered secure for modern cryptographic applications.
All clients are encouraged to adopt the `PBKDF2WithHmacSHA256` algorithm, which provides improved cryptographic strength. 
A fallback mechanism ensures backward compatibility for legacy systems. 

> Updated API Method:

Reviewer: ```suggestion
API Method
```
Also, please change the length of the formatting header lines above and below to match the length of this line after editing.

-------------------

> The following method has been updated to incorporate the new validation mechanism:

Reviewer: ```suggestion
The following method uses the `PBKDF2WithHmacSHA256` validation mechanism and includes a fallback mechanism:
```
Please revise this suggestion if this does not correctly describe your work.
